### PR TITLE
[*] remove prosemirror deps

### PR DESCRIPTION
**Pull request by maimonav**

### CODE REVIEW

File: packages/plugin-table/web/src/tiptap/plugins/controllerPlugin.tsx [hunk 1]
```diff
@@ -14,7 +14,8 @@ export const controllerPlugin = editor =>
       init: () => {
         return {};
       },
-      apply(tr, prev, oldState, newState) {
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
```
Comment (maimonav) [inline packages/plugin-table/web/src/tiptap/plugins/controllerPlugin.tsx:17]: Define type

File: packages/editor-common/web/src/Utils/tiptap/isTextSelection.ts [hunk 2]
```diff
@@ -0,0 +1,6 @@
+import type { Selection } from 'prosemirror-state';
+
+// replacing isTextSelection of tiptap which is broken
+// https://github.com/ueberdosis/tiptap/issues/2979
+export const isTextSelection = (selection: Selection): boolean =>
+  selection.toJSON().type === 'text';
```
Comment (lxgreen) [inline packages/editor-common/web/src/Utils/tiptap/isTextSelection.ts:6]: why not `instanceOf TextSelection`?

Comment (lxgreen) [reply packages/editor-common/web/src/Utils/tiptap/isTextSelection.ts:6]: also, why editor-common? we want to deprecate it at all. how about ricos-common?

**Review by lxgreen — APPROVED** [review]
we saw it all basically. i skipped the table changes for now

File: packages/editor-common/web/src/Utils/tiptap/isTextSelection.ts (hunk 2 above)
Comment (maimonav) [reply packages/editor-common/web/src/Utils/tiptap/isTextSelection.ts:6]: 1. instanceof TextSelection uses prosemirror-state, which costs something like 60KB
https://github.com/wix/ricos/runs/7861688823?check_suite_focus=true#step:7:66
2. We can easy change it to ricos-common when we deprecate editor-common, now it maybe be resulted in bundle increment and I would like to merge it soon.

**Pull request merged**